controllers/ibmpowervsmachine_controller_test.go

```diff
@@ -492,6 +497,303 @@ func TestIBMPowerVSMachineReconciler_ReconcileOperations(t *testing.T) {
 			g.Expect(machineScope.IBMPowerVSMachine.Finalizers).To(ContainElement(infrav1beta2.IBMPowerVSMachineFinalizer))
 			expectConditions(g, machineScope.IBMPowerVSMachine, []conditionAssertion{{infrav1beta2.InstanceReadyCondition, corev1.ConditionFalse, capiv1beta1.ConditionSeverityError, infrav1beta2.InstanceProvisionFailedReason}})
 		})
+
+		t.Run("Should fail reconcile if creation of the load balancer pool member is unsuccessful", func(t *testing.T) {
+			g := NewWithT(t)
+			setup(t)
+			t.Cleanup(teardown)
+			secret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						capiv1beta1.ClusterNameLabel: "powervs-cluster",
+					},
+					Name:      "bootsecret",
+					Namespace: "default",
+				},
+				Data: map[string][]byte{
+					"value": []byte("user data"),
+				},
+			}
+
+			pvsmachine := &infrav1beta2.IBMPowerVSMachine{
+
+				ObjectMeta: metav1.ObjectMeta{
+					Name:       *ptr.To("capi-test-machine"),
+					Finalizers: []string{infrav1beta2.IBMPowerVSMachineFinalizer},
+				},
+				Status: infrav1beta2.IBMPowerVSMachineStatus{
+					Ready: true,
+				},
+				Spec: infrav1beta2.IBMPowerVSMachineSpec{
+					MemoryGiB:  8,
+					Processors: intstr.FromString("0.5"),
+					Image: &infrav1beta2.IBMPowerVSResourceReference{
+						ID: ptr.To("capi-image-id"),
+					},
+					Network: infrav1beta2.IBMPowerVSResourceReference{
+						ID: ptr.To("capi-net-id"),
+					},
+					ServiceInstanceID: *ptr.To("service-instance-1"),
+				},
+			}
+
+			machine := &capiv1beta1.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "owner-machine",
+					Namespace: "default",
+					Labels: map[string]string{
+						"cluster.x-k8s.io/control-plane": "true",
+					},
+				},
+
+				Spec: capiv1beta1.MachineSpec{
+					Bootstrap: capiv1beta1.Bootstrap{
+						DataSecretName: ptr.To("bootsecret"),
+					},
+				},
+			}
+
+			mockclient := fake.NewClientBuilder().WithObjects([]client.Object{secret, pvsmachine, machine}...).Build()
+			machineScope = &scope.PowerVSMachineScope{
```

Comment:
Can't we call NewPowerVSMachineScope()?

Reply:
Correct me if I’m wrong, but based on issue #1937, it’s preferred for UT not to rely on the New functions, as their flow shouldn't be within the scope of the parent test.

Reply:
ok, It would be good if can reduce the number of duplicate lines if possible.

Reply:
reduced overall lines of code